ui/app/components/totp/key-create.hbs

```diff
@@ -3,39 +3,30 @@
   SPDX-License-Identifier: BUSL-1.1
 ~}}
 
-<form {{on "submit" @create}}>
+<form {{on "submit" (perform @onSubmit)}}>
   <div class="box is-sideless is-fullwidth is-marginless">
     <MessageError @model={{@model}} />
-    <NamespaceReminder @mode={{@mode}} @noun="TOTP key" />
-
-    {{#each (if @model.generate @generatedFields @nonGeneratedFields) as |field|}}
+    {{#each @keyFormFields as |field|}}
       {{#let (find-by "name" field @model.allFields) as |attr|}}
-        {{! TODO add model validations }}
-        <FormField @attr={{attr}} @model={{@model}} data-test-field />
+        <FormField @attr={{attr}} @model={{@model}} @modelValidations={{@modelValidations}} />
       {{/let}}
     {{/each}}
 
-    <Totp::KeyCreateToggleGroups @model={{@model}} />
+    <Totp::KeyCreateToggleGroups @model={{@model}} @modelValidations={{@modelValidations}} @groups={{@groups}} />
   </div>
   <div class="field is-grouped-split box is-fullwidth is-bottomless">
     <Hds::ButtonSet>
-      <Hds::Button @text={{if (eq @mode "create") "Create key" "Save"}} type="submit" data-test-totp-create />
-      {{#if (eq @mode "create")}}
-        <Hds::Button
-          @text="Cancel"
-          @color="secondary"
-          @route="vault.cluster.secrets.backend.list-root"
-          @model={{@model.backend}}
-          @query={{hash tab="key"}}
-        />
-      {{else}}
-        <Hds::Button
-          @text="Cancel"
-          @color="secondary"
-          @route="vault.cluster.secrets.backend.show"
-          @models={{array @model.backend @model.id}}
-        />
-      {{/if}}
+      <Hds::Button @text="Create key" type="submit" data-test-totp-create />
```

Comment:
might be useful to rename these selectors so you can reuse the general ones [here](https://github.com/hashicorp/vault/blob/main/ui/tests/helpers/general-selectors.ts#L97-L99)
```suggestion
      <Hds::Button @text="Create key" type="submit" data-test-save />
```